app/notebook-store.js

```diff
@@ -2,133 +2,53 @@
-                return reject(err);
-            }
-
-            stats.fullpath = nbpath;
-
-            if (stats.isDirectory()) {
-                // check if this directory contains an index dashboard
-                exists(path.join(nbpath, INDEX_NB_NAME)).then(
-                    function success(exists) {
-                        stats.isDashboard = stats.hasIndex = exists;
-                        if (stats.hasIndex) {
-                            // check if bundled dashboard has an 'urth_components' dir
-                            fs.stat(path.join(nbpath, 'urth_components'),
-                                function(err, urth_stats) {
-                                    stats.supportsDeclWidgets = !err && urth_stats.isDirectory();
-                                    resolve(stats);
-                                });
-                        } else {
-                            resolve(stats);
-                        }
-                    },
-                    function failure(err) {
-                        reject(err);
-                    }
-                );
-            } else {
-                // might be a dashboard file or a "regular" file
-                stats.isDashboard = stats.isFile() && path.extname(nbpath) === DB_EXT;
-                resolve(stats);
-            }
-        });
-    });
-}
+//////////////////
+// READ OPERATIONS
+//////////////////
 
 // Read notebook from cache or from file
 function _loadNb(nbpath, stats) {
     if (!stats) {
-        stats = stat(nbpath);
+        stats = nbfs.stat(nbpath);
     }
     return Promise.resolve(stats)
         .then(function(stats) {
-            var nbfile;
-            if (stats.hasIndex) {
-                nbfile = path.join(nbpath, INDEX_NB_NAME);
-            } else {
-                nbfile = _appendExt(nbpath);
-            }
-
+            var nbfile = stats.fullpath ||
+                path.join(DATA_DIR, appendExt(nbpath, DB_EXT));
             return new Promise(function(resolve, reject) {
-                var nbAbsPath = path.join(DATA_DIR, nbfile);
-                fs.readFile(nbAbsPath, 'utf8', function(err, rawData) {
+                fs.readFile(nbfile, ENCODING, function(err, rawData) {
                     if (err) {
-                        reject(new Error('Error loading notebook: ' + err.message));
+                        reject(err);
                     } else {
-                        try {
-                            var nb = JSON.parse(rawData);
-                            // cache notebook for future reads -- use given `nbpath` since that
-                            // is path from request. later calls will look up using request path.
-                            store[nbpath] = nb; 
-                            resolve(nb);
-                        } catch(e) {
-                            reject(new Error('Error parsing notebook JSON'));
-                        }
+                        var nb = JSON.parse(rawData);
```

Comment:
Yes, we were already rejecting with a JSON parse error anyway, so I removed the try/catch to clean up the code a bit.